pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -823,3 +825,87 @@ func (r *reconcilerBase) patchSyncMetadata(ctx context.Context, rs client.Object
 	r.logger(ctx).Info("Sync metadata update successful")
 	return true, nil
 }
+
+func (r *reconcilerBase) requeueRSync(ctx context.Context, obj client.Object, rsRef types.NamespacedName) []reconcile.Request {
+	r.logger(ctx).Info(fmt.Sprintf("Changes to %s triggered a reconciliation for the %s (%s).",
+		kinds.ObjectSummary(obj), r.syncKind, rsRef))
+	return []reconcile.Request{
+		{NamespacedName: rsRef},
+	}
+}
+
+func (r *reconcilerBase) requeueAllRSyncs(ctx context.Context, obj client.Object) []reconcile.Request {
+	allRSyncs := r.listRSyncs(ctx)
+	requests := make([]reconcile.Request, len(allRSyncs))
+	for i, rs := range allRSyncs {
+		requests[i] = reconcile.Request{
+			NamespacedName: client.ObjectKeyFromObject(rs),
+		}
+	}
+	if len(requests) > 0 {
+		r.logger(ctx).Info(fmt.Sprintf("Changes to %s triggered reconciliations for %d %s objects.",
+			kinds.ObjectSummary(obj), len(allRSyncs), r.syncKind))
+	}
+	return requests
+}
+
+func (r *reconcilerBase) listRSyncs(ctx context.Context, opts ...client.ListOption) []client.Object {
+	switch r.syncKind {
+	case configsync.RootSyncKind:
+		return r.listRootSyncs(ctx, opts...)
+	case configsync.RepoSyncKind:
+		return r.listRepoSyncs(ctx, opts...)
+	default:
+		panic(fmt.Sprintf("invalid %T.syncKind: %s", r, r.syncKind))
+	}
+}
+
+func (r *reconcilerBase) listRootSyncs(ctx context.Context, opts ...client.ListOption) []client.Object {
+	rsList := &v1beta1.RootSyncList{}
```

Comment:
Should these use `PartialObjectMetadataList` since it's only using the name/namespace of each RSync?

Reply:
Great idea. Now I just need to figure out why that breaks a bunch of unit tests...

Reply:
Ahh. The Fake client doesn't handle conversions to and from PartialObjectMetadataList...

Reply:
I had to do major surgery to the fake client storage to handle partial objects. I'll pull that out to a separate PR tomorrow.

Reply:
rebased.